### PR TITLE
[Package Add] steamtinkerlaunch-git

### LINKED ISSUE #474: PacReq: SteamTinkerLaunch
Steam Tinker Launch (short stl) is a Linux wrapper tool for use with the Steam client which allows customizing and start tools and options for games quickly on the fly

URL: https://github.com/frostworx/steamtinkerlaunch
Building Guide: https://github.com/frostworx/steamtinkerlaunch/blob/master/README.md
Requested by me

### PR DESCRIPTION
**Pull request by echometerain**
resolves #474

### CODE REVIEW

File: packages/steamtinkerlaunch-git/steamtinkerlaunch-git.pacscript [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+gives="steamtinkerlaunch"
+description="Linux wrapper tool for use with the Steam client for custom launch options and 3rd party programs."
+optdepends=("strace: system call tracer"
+	"gamemode: linux gaming optimizer"
+	"mangohud: vulkan/opengl monitoring hud"
+	"winetricks: wine settings tweaker"
+	"vkbasalt: vulkan graphics enhancer"
+	"goverlay: vulkan overlay configurator"
+	"scummvm: VM for point-and-click games"
+	"gameconqueror: memory editing for cheating")
```
Comment (Henryws) [inline packages/steamtinkerlaunch-git/steamtinkerlaunch-git.pacscript:16]: 🤨

Comment (echometerain) [reply packages/steamtinkerlaunch-git/steamtinkerlaunch-git.pacscript:16]: gameconqueror was in the optional dependencies provided by the stl installation guide https://github.com/frostworx/steamtinkerlaunch/wiki/Installation

File: packages/steamtinkerlaunch-git/steamtinkerlaunch-git.pacscript [hunk 2]
```diff
@@ -0,0 +1,44 @@
+name="steamtinkerlaunch-git"
+url="https://github.com/frostworx/steamtinkerlaunch.git"
+build_depends="vim gcc"
+depends="git make gawk xdotool wget vim xdotool x11-utils"
+breaks="steamtinkerlaunch"
```
Comment (Henryws) [inline packages/steamtinkerlaunch-git/steamtinkerlaunch-git.pacscript:5]: add all the breaks

File: packages/steamtinkerlaunch-git/steamtinkerlaunch-git.pacscript [hunk 3]
```diff
@@ -2,7 +2,7 @@ name="steamtinkerlaunch-git"
 url="https://github.com/frostworx/steamtinkerlaunch.git"
 build_depends="vim gcc"
 depends="git make gawk xdotool wget vim xdotool x11-utils"
-breaks="steamtinkerlaunch"
+breaks="steamtinkerlaunch, ${name}-git, ${name}-deb, ${name}-app, ${name}-deb, ${name}-bin"
```
Comment (Henryws) [inline packages/steamtinkerlaunch-git/steamtinkerlaunch-git.pacscript:5]: no commas

File: packages/steamtinkerlaunch-git/steamtinkerlaunch-git.pacscript [hunk 4]
```diff
@@ -0,0 +1,44 @@
+name="steamtinkerlaunch-git"
+url="https://github.com/frostworx/steamtinkerlaunch.git"
+build_depends="vim gcc"
```
Comment (Henryws) [inline packages/steamtinkerlaunch-git/steamtinkerlaunch-git.pacscript:3]: vim?

Comment (echometerain) [reply packages/steamtinkerlaunch-git/steamtinkerlaunch-git.pacscript:3]: oh wait I forgot to change that

**Review by Henryws — APPROVED** [review]

**Comment by wizard-28** [comment]
**DO NOT MERGE TILL #537 IS MERGED**

**Review by wizard-28 — CHANGES_REQUESTED** [review]
Follow the instructions to setup your pacscript development environment listed [here](https://github.com/pacstall/pacstall-programs#how-to-setup-the-environment-for-pacscript-development)

After doing so try staging (`git add`) this file (make a small dummy modification) then try to commit it (after running `pre-commit install`). When the pre-commit hooks fail (if they fail), fix the errors.

File: packages/steamtinkerlaunch-git/steamtinkerlaunch-git.pacscript [hunk 5]
```diff
@@ -0,0 +1,43 @@
+name="steamtinkerlaunch-git"
+url="https://github.com/frostworx/steamtinkerlaunch.git"
+depends="git make gawk xdotool wget vim xdotool x11-utils"
```
Comment (wizard-28) [inline packages/steamtinkerlaunch-git/steamtinkerlaunch-git.pacscript:3]: how does it depend on `git` & `make`, those probably belong in `build_depends`

File: packagelist [hunk 6]
```diff
@@ -164,6 +164,8 @@ spacecadetpinball-git
 spotify-client-deb
 st-distrotube-git
 st-lukesmith-git
+steamtinkerlaunch-git
+stellarium
```
Comment (wizard-28) [inline packagelist:168]:
```suggestion
```

Duplicate

**Review by wizard-28 — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml [hunk 7]
```diff
@@ -3,7 +3,7 @@ ci:
 
 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v3.2.0
+    rev: v4.0.1
```
Comment (wizard-28) [inline .pre-commit-config.yaml:6]: No need to do this automatically, I added a CI, it'll take care of this file automatically from now on

**Review by wizard-28 — APPROVED** [review]

**Pull request merged**